feedback-module/src/components/common/Module.js

```diff
@@ -1,37 +1,89 @@
 import React, { useEffect, useState } from "react";
-import { GridContainer } from "@trussworks/react-uswds";
+import { GridContainer, Grid, Form } from "@trussworks/react-uswds";
 
-import { MODULE_CONTAINER_STYLE } from "../../assets/styling_classnames";
+import {
+  MODULE_CONTAINER_STYLE,
+  SCREEN_CONTAINER_STYLE,
+  H1_STYLE,
+  H1_WHITE_STYLE,
+  FORM_STYLE,
+} from "../../assets/styling_classnames";
+import { SCREENS } from "../../assets/constants";
 import Header from "./Header";
-import Screen1 from "../pages/Screen1";
-import Screen2 from "../pages/Screen2";
-import Screen3 from "../pages/Screen3";
-import Screen4 from "../pages/Screen4";
+import ModuleButton from "./Button";
+import CheckboxList from "../CheckboxList";
+import TextboxList from "../TextboxList";
 
 function Module() {
   const [feedback, setFeedback] = useState({});
+  const [userInfo, setUserInfo] = useState({});
+  const [screen, setScreen] = useState(SCREENS.feedback_type);
+  const [checkedFields, setCheckedFields] = useState(null);
+  const [otherField, setOtherField] = useState("");
+  const [inputQuestions, setInputQuestions] = useState();
 
   useEffect(() => {
+    if (screen.checkboxes) {
+      setCheckedFields(
+        screen.checkboxes.map((checkboxLabel) => {
+          return { label: checkboxLabel, checked: false };
+        })
+      );
+    }
+    if (screen.textInputs) {
+      setInputQuestions(
+        screen.textInputs.map((question) => {
+          return { question: question.text, answer: "" };
+        })
+      );
+    }
     console.log(feedback);
-  }, [feedback]);
-
-  const [screen, setScreen] = useState(
-    <Screen1
-      setFeedback={setFeedback}
-      changePage={(data) =>
-        setScreen(
-          <Screen2
-            feedbackType={data}
-            setFeedback={setFeedback}
-            changePage={() =>
-              setScreen(<Screen3 changePage={() => setScreen(<Screen4 />)} />)
-            }
-          />
-        )
-      }
-      page="[this page]"
-    />
-  );
+    console.log(userInfo);
+  }, [screen]);
+
+  const handleSubmit = (formID) => {
+    const checkedOptions =
+      checkedFields &&
+      checkedFields.map(({ label, checked }) => {
+        return {
+          checked: checked,
+          label: label === "Other" && checked ? `Other: ${otherField}` : label,
+        };
+      });
+
+    setCheckedFields(checkedOptions);
+    if (formID === "feedback") {
+      setFeedback((feedback) => {
+        feedback.checkedOptions = checkedOptions
+          .filter(({ checked }) => checked)
+          .map(({ label }) => label);
+        feedback.inputResponses = inputQuestions;
+        return feedback;
+      });
+    } else if (formID === "research") {
+      setUserInfo(inputQuestions);
+    }
+  };
+
+  const changeScreen = (text, nextScreen, feedbackID) => {
+    if (feedbackID) {
+      setFeedback((feedback) => {
```

Comment:
We might want to change some of these variable names as it will be very hard to understand all this code for someone who has never seen it before. For example maybe instead of setFeedback we call it something like setFeedbackForJSON or setFeedbackForAPI or something along those lines (so they understand what we are keeping track of and why)